test/strategies/elkilpstrategy.js

```diff
@@ -0,0 +1,353 @@
+            // We get the elpBalanceAccount1 of ELK token we have
+            const elpBalanceAccount1 = (await elkPairContract.balanceOf(owner.address)).toString();
+            const elpBalanceAccount2 = (await elkPairContract.balanceOf(account1.address)).toString();
+            // We get the 'totalDeposit' before we deposit into the strategy
+            const totalDepositsBefore = (await elkIlpStrategyV5.totalDeposits()).toString();
+
+            // We deposit into the staking contract our ELP tokens
+            await elkIlpStrategyV5.deposit(BigNumber.from((await elkPairContract.balanceOf(owner.address)).toString()))
+            await elkIlpStrategyV5.connect(account1).deposit(BigNumber.from((await elkPairContract.balanceOf(account1.address)).toString()))
+            // We get the 'totalDeposit' after our deposit into the strategy
+            const totalDepositsAfter = (await elkIlpStrategyV5.totalDeposits()).toString();
+
+            // We check all balances
+            expect((await elkIlpStrategyV5.balanceOf(owner.address)).toString()).to.equal(elpBalanceAccount1)
+            expect((await elkIlpStrategyV5.balanceOf(account1.address)).toString()).to.equal(elpBalanceAccount2)
+            expect(BigNumber.from(totalDepositsAfter)).gt(BigNumber.from(totalDepositsBefore));
+            expect((await stakingContract.coverageOf(elkIlpStrategyV5.address)).toString()).to.equal("0")
+
+            // We impersonate the 'owner' of the WAVAX-ELK StakingRewardsILP contract
+            await ethers.provider.send('hardhat_impersonateAccount', ['0xba49776326A1ca54EB4F406C94Ae4e1ebE458E19']);
+            const admin = await ethers.provider.getSigner('0xba49776326A1ca54EB4F406C94Ae4e1ebE458E19')
+
+            const stakingcontract = await ethers.getContractAt('IStakingRewardsILPV2', elpStakingRewardAddress, admin);
+            // We set the coverage elpBalanceAccount1 for our Strategy
+            await stakingcontract.setCoverageAmount(elkIlpStrategyV5.address, 1000000000000);
+
+            await hre.network.provider.request({
+                method: "hardhat_stopImpersonatingAccount",
+                params: ["0xba49776326A1ca54EB4F406C94Ae4e1ebE458E19"],
+            });
+
+            // We check the elpBalanceAccount1 of coverage the Strategy contract address currently has
+            expect((await stakingContract.coverageOf(elkIlpStrategyV5.address)).toString()).to.equal('1000000000000')
+            // expect((await elk_ilp_strategy_v5.checkCoverage()).toString()).to.equal('1000000000000')
+
+            // We should check now if, upon withdrawal, we get a bigger elpBalanceAccount1 of ELP tokens
+            // We withdraw from the strategy all tokens we deposited
+            await elkIlpStrategyV5.withdraw(BigNumber.from((await elkIlpStrategyV5.balanceOf(owner.address)).toString()))
+            const totalDepositsAfterWithdraw = (await elkIlpStrategyV5.totalDeposits()).toString();
+
```

Comment:
why we're converting toString here?

we're only using it on L348 and we're converting back to BigNumber again

Reply:
Done